signing/interface.go

```diff
@@ -0,0 +1,116 @@
+package signing
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/reddit/baseplate.go/secrets"
+)
+
+// The Version of the signing protocol.
+type Version uint8
+
+// SignArgs are the args passed into Version.Sign function.
+type SignArgs struct {
+	// The message to sign. Required.
+	Message []byte
+
+	// The secret used to sign the message. Required.
+	Key secrets.Secret
+
+	// Signature expiring time.
+	//
+	// V1: If ExpiresAt is non-zero, it will be used.
+	// Otherwise time.Now().Add(ExpiresIn) will be used.
+	// Note that V1 only defined second precision,
+	// any sub-second precision in ExpiresIn or ExpiresAt will be dropped and
+	// rounded down.
+	ExpiresAt time.Time
+	ExpiresIn time.Duration
```

Comment:
Why is there both `ExpiresAt` and `ExpiresIn`?  Wouldn't it be clearer to pick one?

Reply:
So it's easier to use :) I think the documentation clearly explained the priority when both are set, so that should be good enough?

Reply:
It's up to you ultimately, personally I find it better to only have one option rather than two that do the same thing.  It makes it easier to understand the API just by browsing it rather than having to dive into the documentation and it's one less thing to have to check and remember when using it.